Feature/loginpage validation

noelvandevrie opened the pull request:
Fixes: 
Login form validations

Improves:
updated styling of the form/fields to be in line with the passwordreset page

Closes issue #34

noelvandevrie committed:
Added validation and styling to the login fields.

noelvandevrie committed:
Added validation and styling to the login fields.

JustSamuel commented on src/views/LoginView.vue:
```diff
@@ -96,6 +162,7 @@ form {
   display: flex;
   flex-direction: column;
 }
+
```
Tag even ergens hier een issue voor de hoeveelheid CSS.

noelvandevrie replied:
added TODO with issue #  in commit https://github.com/GEWIS/sudosos-frontend-vue3/pull/31/commits/c0cd5afae61160be4c481f76a38752281f930aa6

JustSamuel commented on src/views/LoginView.vue:
```diff
@@ -70,17 +137,16 @@ onBeforeMount(() => {
     if (authStore.getUser) userStore.fetchCurrentUserBalance(authStore.getUser.id, apiService);
     router.push({ name: 'home' });
   }).catch((error) => {
     console.error(error);
```
Issue voor iets beters dan dan gwn log and ignore?

CodeNamedRobin replied:
Zie #18 , wel even een comment plaatsen met TODO en de issue, voor den duidelijkheid.

noelvandevrie replied:
added TODO and issue # in commit https://github.com/GEWIS/sudosos-frontend-vue3/pull/31/commits/c0cd5afae61160be4c481f76a38752281f930aa6

JustSamuel commented on src/views/LoginView.vue:
```diff
@@ -70,17 +137,16 @@ onBeforeMount(() => {
-};
+});
+
 const loginViaGEWIS = () => {
```
Issue maken / taggen dat inloggen via GEWIS nog niet werkt? we doen nog niks met de callback.

CodeNamedRobin replied:
Hier idd een todo maken die #32 benoemd.

noelvandevrie replied:
added TODO and issue # in commit https://github.com/GEWIS/sudosos-frontend-vue3/pull/31/commits/c0cd5afae61160be4c481f76a38752281f930aa6

JustSamuel commented on src/views/LoginView.vue:
```diff
@@ -1,35 +1,94 @@
-      <img id="login-image" src="../assets/img/bier.png" alt="logo" />
+      <img
+          id="login-image"
+          src="../assets/img/bier.png"
```
werkt dit ook als je build? Lijkt op relative routing en dan moet je @ gebruiken. Zie de vite config enzo.

noelvandevrie replied:
This should be fixed now in commit https://github.com/GEWIS/sudosos-frontend-vue3/pull/31/commits/c0cd5afae61160be4c481f76a38752281f930aa6

JustSamuel commented on src/views/LoginView.vue:
```diff
@@ -1,35 +1,94 @@
-          :placeholder="$t('login.Enter password')"
+        <img
+            id="gewis-branding"
+            src="../assets/img/gewis-branding.svg"
```
idem

noelvandevrie replied:
Should be fixed too in commit https://github.com/GEWIS/sudosos-frontend-vue3/pull/31/commits/c0cd5afae61160be4c481f76a38752281f930aa6

JustSamuel commented on src/views/LoginView.vue:
```diff
@@ -1,35 +1,94 @@
+        >
+          {{ $t('login.Username') }}
+        </label>
+        <span class="p-float-label with-error">
```
Voelt alsof we hier een component van kunnen maken? er zit best wat overlap tussen die van de username als die van t password; alleen misschien is een component weer kut met binding enzo. food for though.

CodeNamedRobin replied:
Zou kunnen, maar lijkt me iets voor later, aangezien het me een typisch iets lijkt wat dagen gaat kosten omdat je net niet helemaal weet hoe het in elkaar steekt. Zou hiervoor wel een issue openen.

JustSamuel reviewed (changes requested):
Kleintjes en dan prima.

CodeNamedRobin committed:
Implement all POS management pages. (#16)

* Attempt at fix for product cards

* Added form validation to create container
- Added acutal form to the AddContainerDialogComponent.vue
- Added response to createEmptyContainer method
- Fixed card size for ProductComponent.vue

* Added validation to POSEditView.vue

* Some lint and type-check fixes.

* Small fix from the new common packages.

* Some minor improvements

* Fixed POS Edit page, selectedContainers now actually work.

* Implement adding container functionality

* Fixed form validation resizing input fields

* Some small formatting fixes

* Removed unused console log statements

* Small fixes

* Stylized form validation in adding containers

* Further styling of form validation

* Implemented function to close dialogue on tap anywhere else

---------

Co-authored-by: Samuel <samuelincognito@gmail.com>

CodeNamedRobin commented on src/views/LoginView.vue:
```diff
@@ -70,17 +137,16 @@ onBeforeMount(() => {
-  await authStore.gewisLdapLogin(values.username, values.password, apiService).then(() => {
-    console.error("loggedin");
+const ldapLogin = loginForm.handleSubmit(async () => {
+  if (!username.value.modelValue || !password.value.modelValue) return;
```
Dit is redundant, immers zijn deze values defined (omdat ze required zijn)

noelvandevrie replied:
Removed redundant check for values in commit https://github.com/GEWIS/sudosos-frontend-vue3/pull/31/commits/c0cd5afae61160be4c481f76a38752281f930aa6

CodeNamedRobin commented on src/views/LoginView.vue:
```diff
@@ -135,8 +202,55 @@ main {
+  line-height:18px;
+}
+
+.p-error > i {
```
Als je dan toch veel css gebruikt, doe het dan nested, maakt het wat leesbaarder imo.

noelvandevrie replied:
I feel like this is incorporated in the mentioning of issues #14 and #29

CodeNamedRobin replied:
Fine.

CodeNamedRobin reviewed (changes requested):
Small things.

noelvandevrie committed:
Implemented localLogin as it was missing

CodeNamedRobin committed:
Update README.md

CodeNamedRobin committed:
Added .env-example

CodeNamedRobin committed:
Added license and notice to the repo

noelvandevrie committed:
added TODOs and related them to their respective issues, routing of the images should be fixed now and removed redundant check of the loginForm values.

CodeNamedRobin commented on src/views/LoginView.vue:
```diff
@@ -1,66 +1,133 @@
+        >
+          {{ $t('login.Username') }}
+        </label>
+        <span class="p-float-label with-error">
```
The error message right now moves the whole UI, I'd suggest making it so that it fills a whitespace that is already there.

noelvandevrie replied:
fixed in https://github.com/GEWIS/sudosos-frontend-vue3/pull/31/commits/c9e26c2efbdf6016562d236ad38b95baf1089390

CodeNamedRobin commented on src/views/LoginView.vue:
```diff
@@ -1,66 +1,133 @@
+          {{ $t('login.Username') }}
+        </label>
+        <span class="p-float-label with-error">
+          <InputText
```
Would be nice if this lights up red like in your other PR.

noelvandevrie replied:
fixed in https://github.com/GEWIS/sudosos-frontend-vue3/pull/31/commits/c9e26c2efbdf6016562d236ad38b95baf1089390

CodeNamedRobin reviewed (changes requested):
Small things, mostly cosmetic.

noelvandevrie committed:
Added background error color to the fields and fixed the error messages shifting down elements on the page.

JustSamuel commented on src/views/LoginView.vue:
```diff
@@ -1,66 +1,129 @@
 const authStore = useAuthStore();
 const userStore = useUserStore();
+
 const schema = toTypedSchema(
   yup.object({
```
Vertaling!!!

noelvandevrie replied:
Mentioned in issues #17 and #25

JustSamuel commented on .env:
```diff
@@ -1,2 +1,5 @@
-VITE_APP_API_BASE=https://sudosos.test.gewis.nl/api/v1
-VITE_APP_GEWIS_TOKEN=sudosos-dev
+VITE_APP_API_BASE='https://sudosos.test.gewis.nl/api/v1'
```
mhmmm .env

JustSamuel reviewed (approved):
priem

noelvandevrie committed:
deleted .env 

deleted .env from commit

JustSamuel reviewed (approved):
Mooi.

noelvandevrie committed:
Merge branch 'main' into feature/loginpage-validation

# Conflicts:
#	src/views/LoginView.vue

CodeNamedRobin reviewed (approved):
Mooi.
